### ISSUE TITLE
Doctest failure when running all tests at once

### ISSUE THREAD

**Issue by nmattia** [tweag/rules_haskell#620]
The following error shows up somewhat regularly:

```
[nix-shell:~/projects/tweag/rules_haskell]$ bazel test //tests/...
INFO: Invocation ID: 87a10bac-b772-461b-93c6-e0e5e0199bb8
INFO: Analysed 204 targets (0 packages loaded, 0 targets configured).
INFO: Found 132 targets and 72 test targets...
ERROR: /home/nicolas/projects/tweag/rules_haskell/tests/haskell_doctest/BUILD:40:1: error executing shell command: '/nix/store/pawkx6zfnj59zvpa0hkrrqin9f9iwika-bash/bin/bash -c
        export LANG=en_US.UTF-8
export LOCALE_ARCHIVE=external/glibc_locales/lib/locale/locale-archive
        external/hackage-doctest...' failed (Exit 1) bash failed: error executing command /nix/store/pawkx6zfnj59zvpa0hkrrqin9f9iwika-bash/bin/bash -c ... (remaining 57 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
doctest: /run/user/1001/.doctest-6: createDirectory: already exists (File exists)
```

It can be worked around by specifying `--jobs=1` but this makes running the tests take close to forever.

CC @gdeest

**Comment by nmattia** [comment]
Note: another work around is to comment out those lines: https://github.com/tweag/rules_haskell/blob/f7b355a9c5da4e0b81fc60bc963cd78f0dc4fe0b/tests/BUILD#L174-L181

**Comment by gdeest** [comment]
It looks like `doctest` uses the process' ID to generate its temp directory:

https://github.com/sol/doctest/blob/1cc1902b602ed3c6cf9a1c8c2144acb27435dc17/src/Extract.hs#L146

and brackets around that with createDirectory / removeDirectoryRecursive, so it's a bit confusing that multiple tests (seemingly) attempt to create the same directory.

**Comment by gdeest** [comment]
Hmmm, Linux sandboxing support a mechanism of PID namespacing. This might explain it.

**Comment by gdeest** [comment]
@nmattia Could you try the `fix-doctest` branch and tell me how it fares ?

**Comment by Profpatsch** [comment]
Fixed by #622. If not, please reopen.
